x-pack/legacy/plugins/apm/public/components/app/Settings/AddSettings/AddSettingFlyout.tsx

```diff
@@ -82,8 +84,11 @@ export function AddSettingsFlyout({
     env =>
       env.name === environment && (Boolean(selectedConfig) || env.available)
   );
-  const isSampleRateValid = sampleRate >= 0 && sampleRate <= 1;
-
+  const sampleRateFloat = parseFloat(sampleRate || '');
+  const isSampleRateValid =
+    sampleRateFloat >= 0 &&
+    sampleRateFloat <= 1 &&
+    sampleRateFloat - parseFloat(sampleRateFloat.toFixed(3)) === 0;
```

Comment:
Is this to ensure a maximum of three decimals? It's probably a good idea to extract it and give it a sensible name.

```js
const correctDecimals = Number.isInteger(sampleRateFloat * 1000)
```

Or if we want to make it even more clear:
```js
function countDecimals(value: string) {
  if (!value.includes('.')) {
    return 0;
  }
  return value.split('.')[1].length;
}
```

```js
const correctDecimals = countDecimals() < 4
```